tests/telemetry/test_telemetry.py

```diff
@@ -143,10 +164,41 @@ def test_virtualdb_table_streaming(duthosts, enum_rand_one_per_hwsku_hostname, p
     if duthost.is_supervisor_node():
         pytest.skip("Skipping test as no Ethernet0 frontpanel port on supervisor")
     skip_201911_and_older(duthost)
-    cmd = generate_client_cli(duthost=duthost, gnxi_path=gnxi_path, method=METHOD_SUBSCRIBE, update_count = 3)
+    cmd = generate_client_cli(duthost=duthost, gnxi_path=gnxi_path, method=METHOD_SUBSCRIBE, update_count=3)
     show_gnmi_out = ptfhost.shell(cmd)['stdout']
     result = str(show_gnmi_out)
 
-    assert_equal(len(re.findall('Max update count reached 3', result)), 1, "Streaming update count in:\n{0}".format(result))
-    assert_equal(len(re.findall('name: "Ethernet0"\n', result)), 4, "Streaming updates for Ethernet0 in:\n{0}".format(result)) # 1 for request, 3 for response
-    assert_equal(len(re.findall('timestamp: \d+', result)), 3, "Timestamp markers for each update message in:\n{0}".format(result))
+    assert_equal(len(re.findall('Max update count reached 3', result)), 1,
+                 "Streaming update count in:\n{0}".format(result))
+    assert_equal(len(re.findall('name: "Ethernet0"\n', result)), 4,
+                 "Streaming updates for Ethernet0 in:\n{0}".format(result))  # 1 for request, 3 for response
+    assert_equal(len(re.findall('timestamp: \\d+', result)), 3,
+                 "Timestamp markers for each update message in:\n{0}".format(result))
+
+
+def invoke_py_cli_from_ptf(ptfhost, cmd):
+    ptfhost.shell(cmd)
+
+
+@pytest.mark.disable_loganalyzer
+def test_mem_spike(duthosts, rand_one_dut_hostname, ptfhost, gnxi_path):
+    """Test whether memory usage of telemetry container will exceed threshold
+    if python gNMI client continuously creates channels with gNMI server.
+    """
+    logger.info("Starting to test the memory spike issue of telemetry container")
+
+    duthost = duthosts[rand_one_dut_hostname]
+ 
+    cmd = generate_client_cli(duthost=duthost, gnxi_path=gnxi_path, method=METHOD_SUBSCRIBE,
+                              xpath="DOCKER_STATS", target="STATE_DB", update_count=1, create_connections=2000)
+    client_thread = threading.Thread(target=invoke_py_cli_from_ptf, args=(ptfhost, cmd,))
+    client_thread.start()
+
+    for i in range(MEMORY_CHECKER_CYCLES):
+        ret = duthost.shell("python3 /usr/bin/memory_checker telemetry 419430400", module_ignore_errors=True)
+        assert ret["rc"] == 0, "Memory utilization has exceeded threshold"
+        time.sleep(MEMORY_CHECKER_WAIT)
+
+    client_thread.join()
+
+    postcheck_critical_processes(duthost, "TELEMETRY")
```

Comment:
Why do you check critical process here? Telemetry container restarted for some reasons?